src/account/PluginManagerInternals.sol

```diff
@@ -99,25 +75,40 @@ abstract contract PluginManagerInternals is IPluginManager {
         _selectorData.allowDefaultValidation = false;
     }
 
-    function _addValidationFunction(bytes4 selector, FunctionReference validationFunction)
-        internal
-        notNullFunction(validationFunction)
-    {
-        // Fail on duplicate validation functions. Otherwise, dependency validation functions could shadow
-        // non-depdency validation functions. Then, if a either plugin is uninstalled, it would cause a partial
-        // uninstall of the other.
-        if (!getAccountStorage().validationData[validationFunction].selectors.add(toSetValue(selector))) {
-            revert ValidationFunctionAlreadySet(selector, validationFunction);
+    function _addValidationFunction(address plugin, ManifestValidation memory mv) internal {
+        AccountStorage storage _storage = getAccountStorage();
+
+        FunctionReference validationFunction = FunctionReferenceLib.pack(plugin, mv.functionId);
+
+        if (mv.isDefault) {
+            _storage.validationData[validationFunction].isDefault = true;
+        }
+
+        if (mv.isSignatureValidation) {
```

Comment:
Do we want to keep validation installation in plugin installation? If not, not worth adding new code in.

We talked about only keeping it in one place. While experimenting with validationId, validation installation within plugin installation is almost impossible.
* It require unique onInstall data (an extra validationId) that is not required by other plugin installation.
* We might generate false positive validations due to unable to match the validationIds for selectors actually just using the same validation.

Reply:
I think it's worth considering removing validation install via the manifest completely - this PR doesn't do that yet though, it just removes dependencies.

Reply:
I meant that we don't need to add more code in here like checking `isSignatureValidation` since we plan to remove validation install here altogether. No big deal either way.